### ISSUE TITLE
Gregorian era identifier: ad or ce?

### ISSUE THREAD

**Issue by sffc** [unicode-org/icu4x#470]
On the business logic layer, it will be necessary to have a string identifier for the modern era and pre-modern era of the Gregorian calendar.  Given the decision on the Japanese era (https://github.com/tc39/proposal-temporal/issues/526), where we went with an English transliteration as the identifier string, it seems we have two main choices for the Gregorian eras:

1. `"ad"` and `"bc"`
2. `"ce"` and `"bce"`

For the purposes of formatting and locale data lookup, we will need to disambiguate based on user preference, but that should happen very low in the formatting step.  The date time input model, on the other hand, should not vary based on user preference.

@pedberg-icu @yumaoka @Manishearth @justingrant thoughts?

**Comment by justingrant** [comment]
AD/BC is used more in colloquial/common usage while CE/BCE is used more in academia (history, archaeology, etc.)  

The TL;DR is that academics keep trying to get the general public to use CE/BCE, and the general public seems to consistently say "meh". So there's a lot of back-and forth between experts and non-experts. [Wikipedia](https://en.wikipedia.org/wiki/Common_Era) has a good summary of this dynamic: 

> Some academics in the fields of theology, education, archaeology and history have adopted CE and BCE notation, although there is some disagreement.[49] Several style guides now prefer or mandate its use.[50] The style guide for the Episcopal Diocese Maryland Church News says that BCE and CE should be used.[51]
>
> In the United States, the use of the BCE/CE notation in textbooks was reported in 2005 to be growing.[45] Some publications have moved over to using it exclusively. For example, the 2007 World Almanac was the first edition to switch over to BCE/CE, ending a period of 138 years in which the traditional BC/AD dating notation was used. BCE/CE is used by the College Board in its history tests,[52] and by the Norton Anthology of English Literature. Others have taken a different approach. The US-based History Channel uses BCE/CE notation in articles on non-Christian religious topics such as Jerusalem and Judaism.[53]
>
> In 2002, an advisory panel for the religious education syllabus for England and Wales recommended introducing BCE/CE dates to schools,[54] and by 2018 some local education authorities were using them.[55] In 2018, the National Trust said it would continue to use BC/AD as its house style.[55] English Heritage explains its era policy thus: "It might seem strange to use a Christian calendar system when referring to British prehistory, but the BC/AD labels are widely used and understood."[56] Some parts of the BBC use BCE/CE, but some presenters have said they will not.[55] As at October 2019, the BBC News style guide has entries for AD and BC, but not for CE or BCE.[57]
>
> In June 2006, in the United States, the Kentucky State School Board reversed its decision to use BCE and CE in the state's new Program of Studies, leaving education of students about these concepts a matter of discretion at the local level.[58][59][60]
>
> Also in 2011, media reports suggested that the BC/AD notation in Australian school textbooks would be replaced by BCE/CE notation.[61] The story became national news and drew opposition from some politicians and church leaders. Weeks after the story broke, the Australian Curriculum, Assessment and Reporting Authority denied the rumour and stated that the BC/AD notation would remain, with CE and BCE as an optional suggested learning activity.[62]
>
> In 2013 the Canadian Museum of Civilization (now the Canadian Museum of History) in Ottawa, which had previously switched to BCE/CE, decided to change back to BC/AD in material intended for the public, while retaining BCE/CE in academic content.[63]
>
> The style guide for The Guardian says, under the entry for CE/BCE: "some people prefer CE (common era, current era, or Christian era) and BCE (before common era, etc) to AD and BC, which, however, remain our style".[64]

I'd frame the decision like this:
* If we want maximum name recognition and ease of use from global or self-taught developers without fancy academic credentials, then bc/ad is probably best
* If we want to align with academic and scientific consensus and wait to avoid potential controversy, then ce/bce is probably best

I don't have a strong opinion about this choice. My only strong opinion is that we should clearly document the era codes for all built-in calendars.

**Comment by sffc** [comment]
Thanks for the background.  For the purpose of identifiers, I prefer taking the academic approach.  In #445, I've adopted the `ce`/`bce` convention.  Please note that this is only for the era *identifiers*; the rendered output will still show AD/BC depending on the locale and user preference.

**Comment by Louis-Aime** [comment]
I wonder whether the problem of choosing _codes_ should refer to ideological unresolved debates. CLDR is more concerned with this. IMHO, the codes could be e0 and e1, meaning "ERA0" and "ERA1", the fall-back values of CLDR when the Locale does not give a result. This codes would just suggest that in e0  you count years backwards, as it happens in many (but not all) calendars). If you count with negative years, you do not need another era. And in the case of the ethiopic calendar, we would say "e1" for Amete Ahmet, "creation of world", and "e2" for Amete Mihret (incarnation).
Doing so, intercalendar code would be easier to develop, and philosophical problems would be settled culture by culture.

Anyway, a code remains a code, once fixed, people use it without asking questions. So let's go ahead.

**Comment by sffc** [comment]
2021-02-04: Go with recommendation in https://github.com/unicode-org/icu4x/issues/470#issuecomment-772735333.

**Comment by sffc** [comment]
> IMHO, the codes could be e0 and e1, meaning "ERA0" and "ERA1", the fall-back values of CLDR when the Locale does not give a result.

I agree with this sentiment, but the overwhelming feeling of others, based on the ICU4X design meeting last week, was that a human readable identifier for eras was more desirable.

**Comment by justingrant** [comment]
> For the purpose of identifiers, I prefer taking the academic approach. In #445, I've adopted the `ce`/`bce` convention. Please note that this is only for the era _identifiers_; the rendered output will still show AD/BC depending on the locale and user preference.

As requested, Temporal now uses `ce`/`bce` as era codes.
